examples/network/messaging.rs

```diff
@@ -85,9 +85,12 @@ impl<M: Send> Messaging<M> {
     }
 
     /// Spawns the message delivery thread in a given thread scope.
-    pub fn spawn<'a>(&self, scope: &Scope<'a>) -> ScopedJoinHandle<Result<(), Error>>
+    pub fn spawn<'a, 'scope>(
```

Comment:
If those lifetimes are the same (or even elided) it doesn't work?

Reply:
Let's see. `Scope` has a lifetime parameter, so `'a` is required. Then `ScopedJoinHandle` also now has a lifetime parameter which cannot be the same as `'a` or it won't compile. I think both parameters are unavoidable as long as the function takes a `Scope` and returns a `ScopedJoinHandle`.